lib/solidus_six_saferpay/payment_initialize_params.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+module SolidusSixSaferpay
+  # Provide values for initializing saferpay payment
+  # This class can be overridden to override all or specific values by setting
+  # config.payment_initialize_params_class
+  #
+  #
+  class PaymentInitializeParams
+    include Spree::Tax::TaxHelpers
+
+    attr_reader :order, :payment_method, :return_urls
+
+    def initialize(order, payment_method, return_urls)
+      @order = order
+      @payment_method = payment_method
+      @return_urls = return_urls
+    end
+
+    def params
+      initialize_params = {
+        payment: six_payment,
+        order: six_order,
+        payer: six_payer,
+        return_urls: return_urls
+      }
+
+      six_payment_methods = payment_method.enabled_payment_methods
+      initialize_params[:payment_methods] = six_payment_methods if six_payment_methods.present?
+
+      initialize_params
+    end
+
+    private
+
+    def six_payment
+      six_amount = Spree::Money.new(order.total, currency: order.currency)
+      SixSaferpay::Payment.new(
+        amount: SixSaferpay::Amount.new(value: six_amount.cents, currency_code: six_amount.currency.iso_code),
+        order_id: order.number,
+        description: order.number
+      )
+    end
+
+    def six_address(solidus_address)
+      address_name = extract_name(solidus_address)
+      SixSaferpay::Address.new(
+        first_name: address_name.first_name,
+        last_name: address_name.last_name,
+        date_of_birth: nil,
+        company: nil,
+        gender: nil,
+        legal_form: nil,
+        street: solidus_address.address1,
+        street2: solidus_address.address2,
+        zip: solidus_address.zipcode,
+        city: solidus_address.city,
+        country_subdevision_code: nil,
```

Comment:
Change `country_subdevision_code` to `country_subdivision_code`.

Reply:
Unfortunately this spelling depends on the interface exposed by the [six_saferpay gem](https://github.com/fadendaten/six_saferpay/blob/master/lib/six_saferpay/models/address.rb#L31).
I have opened an issue here: https://github.com/fadendaten/six_saferpay/issues/23